src/util/Util.js

```diff
@@ -11,11 +11,45 @@
  */
 
 /* eslint complexity: [2, 13], max-depth: [2, 3] */
-define(['okta', 'util/Logger'], function (Okta, Logger) {
+define(['okta', './Logger', './Enums'], function (Okta, Logger, Enums) {
 
   var Util = {};
   var _ = Okta._;
 
+  var buildInputForParameter = function (name, value) {
+    var input = document.createElement('input');
+    input.name = name;
+    input.value = decodeURIComponent(value);
```

Comment:
We should probably validate that `value` isn't a malformed URI, otherwise this assignment will throw an error.

Reply:
there are tests in Util_spec to test malformed value. are those what you meant or any examples you might think of?

Reply:
Hmm. Maybe I missed it, but it doesn't look like `Util_spec` is asserting failures for invalid URLs.

I'm guessing this is a fringe case because we're running `decodeURIComponent` on the query string only, but wouldn't the following case would result in a thrown error?

```javascript
setCookieAndRedirect("https://foo.com?q=%E0%A4%A")
```

^ Looking at the code execution, we're going to attempt to `buildDynamicForm` with the encoded value returned from the [`sessionCookieRedirectTpl`](https://github.com/okta/okta-signin-widget/blob/2e662d97ef63a37c55bf2bd90c8af3373a98311a/src/util/RouterUtil.js#L34-L37). Seems like this would error, right?

Of course, let me know if I'm missing something.

Reply:
hmm, what I found is `setCookieAndRedirect("https://foo.com/.....")` would translate to
`redirectFn('http://example.com/?redirectURI=https%3A%2F%2Ffoo.com%3Fq%3D%25E0%25A4%25A')`
which would create a form like
```
'<form method="get" style="display: none;" action="http://example.com">' +
'<input name="redirectURI" type="hidden" value="https://foo.com?q=%E0%A4%A">' +
'</form>'
```
hence the redirect function would not throw any errors and be able to redirect but okta side may not able to interpret the redirect URI.
I tested with `redirectWithFormGet` feature on and off with following code and at both cases are able to redirect back to okta side (though it's error from okta side)

```
res.session.setCookieAndRedirect(signinWidgetOptions.baseUrl + '/app/UserHome?q=%E0%A4%A');
```